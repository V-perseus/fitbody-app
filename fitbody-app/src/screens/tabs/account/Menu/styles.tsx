import { StyleSheet } from 'react-native'
import globals from '../../../../config/globals'

const styles = StyleSheet.create({
  container: {
    ...globals.styles.container,
    alignItems: 'center',
  },
  view: {
    ...globals.styles.view,
    width: '100%',
    padding: 24,
    marginTop: 30,
  },
  headerTitle: {
    ...globals.header.headerTitleStyle,
    color: globals.styles.colors.colorWhite,
  },
  logo: {
    alignSelf: 'center',
    marginTop: 53,
    marginBottom: 20,
  },
  buttonContainer: {
    flexDirection: 'row',
    backgroundColor: globals.styles.colors.colorWhite,
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 8,
    height: 55,
    marginTop: 12,
    marginBottom: 12,
  },
  buttonText: {
    color: globals.styles.colors.colorPink,
    fontSize: 16,
    fontFamily: globals.fonts.primary.semiboldStyle.fontFamily,
  },
  notification: {
    marginLeft: 12,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: globals.styles.colors.colorPink,
    // width: 26,
    paddingHorizontal: 8,
    height: 24,
    borderRadius: 12,
  },
  notificationText: {
    fontSize: 14,
    fontFamily: globals.fonts.primary.semiboldStyle.fontFamily,
    color: globals.styles.colors.colorWhite,
  },
})

export default styles
