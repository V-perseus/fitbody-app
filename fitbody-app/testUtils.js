import React from 'react'
import { render as rtlRender } from '@testing-library/react-native'
import { configureStore } from '@reduxjs/toolkit'
import { NavigationContainer } from '@react-navigation/native'
import { createStackNavigator } from '@react-navigation/stack'
import { Provider } from 'react-redux'

// Import your own reducer
import { appReducer } from './src/store'
import { ModalStackScreens } from './src/config/routes/loggedInStack'

// custom render method override
function render(ui, { preloadedState, store = configureStore({ reducer: appReducer, preloadedState }), ...renderOptions } = {}) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react-native'
// override render method
export { render }

// custom navigation mock
const Stack = createStackNavigator()
export const MockedNavigator = ({
  component,
  params = {},
  preloadedState,
  store = configureStore({ reducer: appReducer, preloadedState }),
}) => {
  return (
    <Provider store={store}>
      <NavigationContainer>
        <Stack.Navigator>
          <Stack.Screen name="MockedScreen" component={component} initialParams={params} />
          <Stack.Screen name="Modals" component={ModalStackScreens} />
        </Stack.Navigator>
      </NavigationContainer>
    </Provider>
  )
}
