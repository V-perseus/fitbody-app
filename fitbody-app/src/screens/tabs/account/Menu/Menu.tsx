import React, { useEffect } from 'react'
import LinearGradient from 'react-native-linear-gradient'
import { Text, TouchableOpacity, View, ScrollView, Pressable } from 'react-native'
import { useDispatch } from 'react-redux'

// Assets
import { translate } from '../../../../../App'
import styles from './styles'
import globals from '../../../../config/globals'
import FitBody from '../../../../../assets/images/logo/fitbody.svg'
import Logout from '../../../../../assets/images/svg/icon/32px/log out.svg'
import Notification from '../../../../../assets/images/svg/icon/32px/notification.svg'

// Components
import FocusAwareStatusBar from '../../../../shared/FocusAwareStatusBar'

// Services
import { clearUser } from '../../../../data/user'
import { clearWorkouts, fetchTrainersPrograms } from '../../../../data/workout'
import { clearNotifications } from '../../../../data/notification'
import { clearSession } from '../../../../services/session'
import { useAppSelector } from '../../../../store/hooks'
import { MenuScreenUseNavigationProp } from '../../../../config/routes/routeTypes'
import { useSegmentLogger } from '../../../../services/hooks/useSegmentLogger'

interface IMenuProps {
  navigation: MenuScreenUseNavigationProp
}
const Menu = ({ navigation }: IMenuProps) => {
  const { navigate } = navigation
  const {
    notification,
    user: { is_ambassador: isAmbassador },
  } = useAppSelector((state) => state.data)

  const unread = notification.filter((item) => item.read_at === null)
  const dispatch = useDispatch()
  const { logEvent } = useSegmentLogger()

  function showModal() {
    if (!isAmbassador) {
      logEvent(null, 'Popup Shown', {
        name: 'Invite A Friend',
        step: 'Join Now',
      })
    } else {
      logEvent(null, 'Popup Shown', {
        name: 'Invite A Friend',
        step: 'Share Your Link',
      })
    }
    navigate('Modals', { screen: 'Invite' })
  }

  function handleContactUs() {
    navigate('Modals', {
      screen: 'ContactDialog',
    })
  }

  useEffect(() => {
    const doLogout = () => {
      clearUser()
      clearNotifications()
      clearWorkouts()
      clearSession()
    }

    const handleLogout = () => {
      navigation.navigate('Modals', {
        screen: 'ConfirmationDialog',
        params: {
          title: 'Are you sure you want to sign out of the app?',
          body: null,
          yesLabel: 'SIGN OUT',
          noLabel: 'NEVER MIND',
          yesHandler: doLogout,
          // noHandler: () => {},
          iconType: null,
        },
      })
    }

    navigation.setOptions({
      headerTitle: () => null,
      headerRight: () => (
        <Pressable onPress={handleLogout} style={{ paddingHorizontal: 24 }}>
          <Logout color={globals.styles.colors.colorWhite} />
        </Pressable>
      ),
      headerLeft: () => (
        <Pressable onPress={() => navigation.navigate('Notifications')} style={{ paddingHorizontal: 24 }} hitSlop={8}>
          <Notification color={globals.styles.colors.colorWhite} />
          {unread.length > 0 && (
            <View
              style={{
                position: 'absolute',
                backgroundColor: globals.styles.colors.colorLove,
                right: 24,
                top: 2,
                borderRadius: 6.5,
                width: 13,
                height: 13,
              }}
            />
          )}
        </Pressable>
      ),
    })
  }, [unread.length, dispatch, navigation])

  // console.log(isAmbassador, ambassadorLink, ambassadorCode)
  return (
    <LinearGradient style={styles.container} colors={[globals.styles.colors.colorPink, globals.styles.colors.colorLavender]}>
      <FocusAwareStatusBar barStyle="light-content" />
      <View style={styles.view}>
        {/* Body Love logo */}
        <View style={styles.logo}>
          <FitBody color={globals.styles.colors.colorWhite} />
        </View>
        <ScrollView showsVerticalScrollIndicator={false}>
          {/* For test builds only. Exposes paywall without needing an expired/non-existing subscription */}
          {/* <TouchableOpacity style={styles.buttonContainer} onPress={() => navigate('Subscription')}>
            <Text style={styles.buttonText}>SUBSCRIBE NOW</Text>
          </TouchableOpacity> */}
          {/* For test builds only, truggers redownload of all trainers and programs */}
          {/* <TouchableOpacity style={styles.buttonContainer} onPress={() => fetchTrainersPrograms({ forceDownload: true })}>
            <Text style={styles.buttonText}>FORCE DOWNLOAD</Text>
          </TouchableOpacity> */}
          {/* Profile */}
          <TouchableOpacity style={styles.buttonContainer} onPress={() => navigate('ProfileView')}>
            <Text style={styles.buttonText}>{translate('app.profile.copy4').toUpperCase()}</Text>
          </TouchableOpacity>

          {/* invite a friend */}
          <TouchableOpacity style={styles.buttonContainer} onPress={showModal}>
            <Text style={styles.buttonText}>{translate('app.profile.copy56').toUpperCase()}</Text>
          </TouchableOpacity>

          {/* Progress Photos */}
          <TouchableOpacity style={styles.buttonContainer} onPress={() => navigate('ProgressPhotos')}>
            <Text style={styles.buttonText}>{translate('app.profile.copy5').toUpperCase()}</Text>
          </TouchableOpacity>

          {/* FAQ */}
          <TouchableOpacity style={styles.buttonContainer} onPress={() => navigate('FAQ')}>
            <Text style={styles.buttonText}>{translate('app.profile.copy1').toUpperCase()}</Text>
          </TouchableOpacity>

          {/* Email Us */}
          <TouchableOpacity
            style={styles.buttonContainer}
            onPress={handleContactUs}
            // onPress={() => Linking.openURL('mailto:hello@fitbodyapp.com?subject=Question about my Fit Body app')}
          >
            <Text style={styles.buttonText}>{translate('app.profile.copy').toUpperCase()}</Text>
          </TouchableOpacity>

          {/* Settings */}
          <TouchableOpacity style={styles.buttonContainer} onPress={() => navigate('Settings')}>
            <Text style={styles.buttonText}>{translate('app.profile.copy6').toUpperCase()}</Text>
          </TouchableOpacity>
        </ScrollView>
      </View>
    </LinearGradient>
  )
}

export default Menu
