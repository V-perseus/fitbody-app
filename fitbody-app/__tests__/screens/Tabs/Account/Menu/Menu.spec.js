import { MockedNavigator, render } from '../../../../../testUtils'

import Menu from '../../../../../src/screens/tabs/account/Menu'

describe('Menu Screen', () => {
  it('renders snapshot correctly', () => {
    const comp = MockedNavigator({ component: Menu })
    const tree = render(comp).toJSON()
    expect(tree).toMatchSnapshot()
  })

  // it('renders all buttons correctly', () => {
  //   const comp = MockedNavigator({ component: Menu })
  //   const { getByText } = render(comp)

  //   const profileBtn = getByText(/profile/i)

  //   expect(profileBtn).not.toBeNull()
  //   // fireEvent.press(profileBtn)
  //   // expect(counterText.props.children).toEqual(['Current count: ', 1])
  //   // fireEvent.press(decrement)
  //   // expect(counterText.props.children).toEqual(['Current count: ', 0])
  // })
})
