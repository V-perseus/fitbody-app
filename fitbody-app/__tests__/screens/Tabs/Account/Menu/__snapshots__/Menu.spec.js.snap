// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Screen renders snapshot correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "zIndex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          onLayout={[Function]}
          pointerEvents="box-none"
          style={null}
        >
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "flex": 1,
                  "shadowColor": "rgb(216, 216, 216)",
                  "shadowOffset": Object {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              Object {
                "height": 44,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-start",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginStart": 0,
                    "opacity": 1,
                  }
                }
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  hitSlop={8}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "paddingHorizontal": 24,
                    }
                  }
                >
                  <notification.svg
                    color="#ffffff"
                  />
                </View>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "justifyContent": "center",
                    "marginHorizontal": 16,
                    "maxWidth": 590,
                    "opacity": 1,
                  }
                }
              />
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                    "opacity": 1,
                  }
                }
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "paddingHorizontal": 24,
                    }
                  }
                >
                  <log out.svg
                    color="#ffffff"
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          gestureResponseDistance={
            Object {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                handlerTag={1}
                handlerType="PanGestureHandler"
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <BVLinearGradient
                        colors={
                          Array [
                            4294610389,
                            4291149311,
                          ]
                        }
                        endPoint={
                          Object {
                            "x": 0.5,
                            "y": 1,
                          }
                        }
                        locations={null}
                        startPoint={
                          Object {
                            "x": 0.5,
                            "y": 0,
                          }
                        }
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "flex": 1,
                              "flexDirection": "column",
                              "marginHorizontal": 15,
                              "marginTop": 30,
                              "padding": 24,
                              "width": "100%",
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignSelf": "center",
                                "marginBottom": 20,
                                "marginTop": 53,
                              }
                            }
                          >
                            <fitbody.svg
                              color="#ffffff"
                            />
                          </View>
                          <RCTScrollView
                            showsVerticalScrollIndicator={false}
                          >
                            <View>
                              <View
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "backgroundColor": "#ffffff",
                                    "borderRadius": 8,
                                    "flexDirection": "row",
                                    "height": 55,
                                    "justifyContent": "center",
                                    "marginBottom": 12,
                                    "marginTop": 12,
                                    "opacity": 1,
                                  }
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "color": "#fa8dd5",
                                      "fontFamily": "OpenSans-Semibold",
                                      "fontSize": 16,
                                    }
                                  }
                                >
                                  SUBSCRIBE NOW
                                </Text>
                              </View>
                              <View
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "backgroundColor": "#ffffff",
                                    "borderRadius": 8,
                                    "flexDirection": "row",
                                    "height": 55,
                                    "justifyContent": "center",
                                    "marginBottom": 12,
                                    "marginTop": 12,
                                    "opacity": 1,
                                  }
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "color": "#fa8dd5",
                                      "fontFamily": "OpenSans-Semibold",
                                      "fontSize": 16,
                                    }
                                  }
                                >
                                  FORCE DOWNLOAD
                                </Text>
                              </View>
                              <View
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "backgroundColor": "#ffffff",
                                    "borderRadius": 8,
                                    "flexDirection": "row",
                                    "height": 55,
                                    "justifyContent": "center",
                                    "marginBottom": 12,
                                    "marginTop": 12,
                                    "opacity": 1,
                                  }
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "color": "#fa8dd5",
                                      "fontFamily": "OpenSans-Semibold",
                                      "fontSize": 16,
                                    }
                                  }
                                >
                                  [MISSING "EN.APP.PROFILE.COPY4" TRANSLATION]
                                </Text>
                              </View>
                              <View
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "backgroundColor": "#ffffff",
                                    "borderRadius": 8,
                                    "flexDirection": "row",
                                    "height": 55,
                                    "justifyContent": "center",
                                    "marginBottom": 12,
                                    "marginTop": 12,
                                    "opacity": 1,
                                  }
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "color": "#fa8dd5",
                                      "fontFamily": "OpenSans-Semibold",
                                      "fontSize": 16,
                                    }
                                  }
                                >
                                  [MISSING "EN.APP.PROFILE.COPY56" TRANSLATION]
                                </Text>
                              </View>
                              <View
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "backgroundColor": "#ffffff",
                                    "borderRadius": 8,
                                    "flexDirection": "row",
                                    "height": 55,
                                    "justifyContent": "center",
                                    "marginBottom": 12,
                                    "marginTop": 12,
                                    "opacity": 1,
                                  }
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "color": "#fa8dd5",
                                      "fontFamily": "OpenSans-Semibold",
                                      "fontSize": 16,
                                    }
                                  }
                                >
                                  [MISSING "EN.APP.PROFILE.COPY5" TRANSLATION]
                                </Text>
                              </View>
                              <View
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "backgroundColor": "#ffffff",
                                    "borderRadius": 8,
                                    "flexDirection": "row",
                                    "height": 55,
                                    "justifyContent": "center",
                                    "marginBottom": 12,
                                    "marginTop": 12,
                                    "opacity": 1,
                                  }
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "color": "#fa8dd5",
                                      "fontFamily": "OpenSans-Semibold",
                                      "fontSize": 16,
                                    }
                                  }
                                >
                                  [MISSING "EN.APP.PROFILE.COPY1" TRANSLATION]
                                </Text>
                              </View>
                              <View
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "backgroundColor": "#ffffff",
                                    "borderRadius": 8,
                                    "flexDirection": "row",
                                    "height": 55,
                                    "justifyContent": "center",
                                    "marginBottom": 12,
                                    "marginTop": 12,
                                    "opacity": 1,
                                  }
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "color": "#fa8dd5",
                                      "fontFamily": "OpenSans-Semibold",
                                      "fontSize": 16,
                                    }
                                  }
                                >
                                  [MISSING "EN.APP.PROFILE.COPY" TRANSLATION]
                                </Text>
                              </View>
                              <View
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "backgroundColor": "#ffffff",
                                    "borderRadius": 8,
                                    "flexDirection": "row",
                                    "height": 55,
                                    "justifyContent": "center",
                                    "marginBottom": 12,
                                    "marginTop": 12,
                                    "opacity": 1,
                                  }
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "color": "#fa8dd5",
                                      "fontFamily": "OpenSans-Semibold",
                                      "fontSize": 16,
                                    }
                                  }
                                >
                                  [MISSING "EN.APP.PROFILE.COPY6" TRANSLATION]
                                </Text>
                              </View>
                            </View>
                          </RCTScrollView>
                        </View>
                      </BVLinearGradient>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
